### PR TITLE
ART-11061 Scan for changes in an image's dependencies

### PR DESCRIPTION
**Pull request by locriandev**
Ref. [ART-11061](https://issues.redhat.com/browse/ART-11061)

### CODE REVIEW

File: doozer/doozerlib/cli/scan_sources_konflux.py [hunk 1]
```diff
@@ -446,6 +450,52 @@ async def scan_for_config_changes(self, image_meta: ImageMetadata):
...
+            if builder.member:
+                dependencies.add(builder.member)
+        self.logger.info('Checking dependencies of %s: %s', image_meta.distgit_key, ','.join(dependencies))
+
+        for dep_key in dependencies:
+            # Is the image dependency included in doozer --images list?
+            if not self.runtime.image_map.get(dep_key, None):
+                self.logger.warning("Image %s has unknown dependency %s. Is it excluded?",
+                                    image_meta.distgit_key, dep_key)
+                return
```
Comment (vfreex) [inline doozer/doozerlib/cli/scan_sources_konflux.py:477]: I think `continue` should be used here to continue to compare other dependencies.

File: doozer/doozerlib/cli/scan_sources_konflux.py [hunk 2]
```diff
@@ -446,6 +450,52 @@ async def scan_for_config_changes(self, image_meta: ImageMetadata):
...
+
+            # Is the dependency ever been built?
+            dependency_build_record = self.latest_image_build_records_map.get(dep_key, None)
+            if not dependency_build_record:
+                self.logger.warning('Dependency %s of image %s has never been built',
+                                    dep_key, image_meta.distgit_key)
+                return
```
Comment (vfreex) [inline doozer/doozerlib/cli/scan_sources_konflux.py:484]: same here

File: doozer/doozerlib/cli/scan_sources_konflux.py [hunk 3]
```diff
@@ -446,6 +450,52 @@ async def scan_for_config_changes(self, image_meta: ImageMetadata):
...
+
+            # Is the dependency build newer than the dependent's one?
+            dep_rebase_time = isolate_timestamp_in_release(dependency_build_record.release)
+            if not dep_rebase_time:  # no timestamp string in NVR?
+                self.logger.warning('Could not determine dependency rebase time from release %s',
+                                    dependency_build_record.release)
+                return
```
Comment (vfreex) [inline doozer/doozerlib/cli/scan_sources_konflux.py:491]: same here

File: doozer/doozerlib/cli/scan_sources_konflux.py [hunk 4]
```diff
@@ -446,6 +450,52 @@ async def scan_for_config_changes(self, image_meta: ImageMetadata):
...
+
+            dep_rebase_time = datetime.strptime(dep_rebase_time, "%Y%m%d%H%M%S").replace(tzinfo=timezone.utc)
+            if dep_rebase_time > rebase_time:
+                self.add_image_meta_change(
+                    image_meta,
+                    RebuildHint(RebuildHintCode.DEPENDENCY_NEWER, 'Dependency has a newer build'))
```
Comment (vfreex) [inline doozer/doozerlib/cli/scan_sources_konflux.py:497]: A `break` can be used here to stop checking remaining dependencies.

**Review by vfreex — COMMENTED** [review]
Generally looks good to me. I added some minor suggestions for loop control.

File: doozer/doozerlib/cli/scan_sources_konflux.py (hunk 1 above)
Comment (locriandev) [reply doozer/doozerlib/cli/scan_sources_konflux.py]: great catch! This used to be an internal function, and `return` survived the refactoring. Updated in 4dd466c700cc47c439d08b0587ddf397cc9b9538

File: doozer/doozerlib/cli/scan_sources_konflux.py (hunk 3 above)
Comment (locriandev) [reply doozer/doozerlib/cli/scan_sources_konflux.py]: Fixed in 4dd466c700cc47c439d08b0587ddf397cc9b9538

File: doozer/doozerlib/cli/scan_sources_konflux.py (hunk 4 above)
Comment (locriandev) [reply doozer/doozerlib/cli/scan_sources_konflux.py:497]: Fixed in 4dd466c700cc47c439d08b0587ddf397cc9b9538

File: doozer/doozerlib/cli/scan_sources_konflux.py (hunk 2 above)
Comment (locriandev) [reply doozer/doozerlib/cli/scan_sources_konflux.py]: Fixed in 4dd466c700cc47c439d08b0587ddf397cc9b9538

**Review by vfreex — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**